project/SparkBuild.scala

```diff
@@ -862,21 +862,27 @@ object SparkConnectClient {
     (assembly / assemblyPackageScala / assembleArtifact) := false,
 
     // Exclude `pmml-model-*.jar`, `scala-collection-compat_*.jar`,`jsr305-*.jar` and
-    // `netty-*.jar` and `unused-1.0.0.jar` from assembly.
+    // `unused-1.0.0.jar` and checker-qual from assembly.
     (assembly / assemblyExcludedJars) := {
       val cp = (assembly / fullClasspath).value
       cp filter { v =>
         val name = v.data.getName
         name.startsWith("pmml-model-") || name.startsWith("scala-collection-compat_") ||
-          name.startsWith("jsr305-") || name.startsWith("netty-") || name == "unused-1.0.0.jar"
+          name.startsWith("jsr305-") || name == "unused-1.0.0.jar" || name.startsWith("checker-qual")
       }
     },
 
     (assembly / assemblyShadeRules) := Seq(
-      ShadeRule.rename("io.grpc.**" -> "org.sparkproject.connect.client.grpc.@0").inAll,
-      ShadeRule.rename("com.google.protobuf.**" -> "org.sparkproject.connect.protobuf.@1").inAll,
-      ShadeRule.rename("com.google.common.**" -> "org.sparkproject.connect.client.guava.@1").inAll,
-      ShadeRule.rename("com.google.thirdparty.**" -> "org.sparkproject.connect.client.guava.@1").inAll,
+      ShadeRule.rename("io.grpc.**" -> "org.sparkproject.connect.client.io.grpc.@1").inAll,
+      ShadeRule.rename("com.google.**" -> "org.sparkproject.connect.client.com.google.@1").inAll,
+      ShadeRule.rename("io.netty.**" -> "org.sparkproject.connect.client.io.netty.@1").inAll,
+      ShadeRule.rename("org.checkerframework.**" -> "org.sparkproject.connect.client.org.checkerframework.@1").inAll,
+      ShadeRule.rename("javax.annotation.**" -> "org.sparkproject.connect.client.javax.annotation.@1").inAll,
```

Comment:
There are only 4 sub dir in `org.sparkproject.connect.client`:

- com	
- common	
- io	
- proto

Some sub dir not found, such as `javax`, `org/codehaus` or `perfmark` found

Reply:
Yeah, currently the sbt and maven result are not exactly the same. SBT seems skipped all runtime dependencies. Need to look a bit more. I assume you got the SBT result, as all above are "runtime" dependencies.

Reply:
> ```
> [error] /home/runner/work/spark/spark/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/DatasetSuite.scala:22:16: object netty is not a member of package io.grpc
> [error] import io.grpc.netty.NettyServerBuilder
> [error]                ^
> [error] /home/runner/work/spark/spark/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/DatasetSuite.scala:52:14: not found: value NettyServerBuilder
> [error]     val sb = NettyServerBuilder
> [error]              ^
> [error] /home/runner/work/spark/spark/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/SparkSessionSuite.scala:20:16: object netty is not a member of package io.grpc
> [error] import io.grpc.netty.NettyServerBuilder
> [error]                ^
> [error] /home/runner/work/spark/spark/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/SparkSessionSuite.scala:38:14: not found: value NettyServerBuilder
> [error]     val sb = NettyServerBuilder
> [error]              ^
> [error] /home/runner/work/spark/spark/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala:22:16: object netty is not a member of package io.grpc
> [error] import io.grpc.netty.NettyServerBuilder
> [error]                ^
> [error] /home/runner/work/spark/spark/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala:39:14: not found: value NettyServerBuilder
> [error]     val sb = NettyServerBuilder
> [error]              ^
> [error] 6 errors found
> ```
> 
> GA failed as above

I can't check the result of maven because of maven compilation failure  now

Reply:
Fixed.